File: app/src/main/java/com/riis/simple/etaandroid/CompanyActivity.java
package com.riis.simple.etaandroid;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class CompanyActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(com.riis.simple.etaandroid.R.layout.activity_company);
        String[] busNamesArray = getResources().getStringArray(com.riis.simple.etaandroid.R.array.busnames);
        ListView busListView = (ListView) findViewById(com.riis.simple.etaandroid.R.id.busses);
        busListView.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, busNamesArray));

        //Go to RouteActivity and pass company
        busListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent intent = new Intent(CompanyActivity.this, RouteActivity.class);
                intent.putExtra(RouteActivity.EXTRA_COMPANY, position + 1);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/kotlin/com.riis.simple.etaandroid/view/CompanyActivity.kt
package com.riis.simple.etaandroid.view

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import com.riis.simple.etaandroid.R

class CompanyActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_company)

        val busListView: ListView = findViewById(com.riis.simple.etaandroid.R.id.busses) as ListView
        val busNamesArray = resources.getStringArray(R.array.busnames)
        busListView.adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, busNamesArray)

        //Go to RouteActivity and pass company
        busListView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            val intent = Intent(this@CompanyActivity, RouteActivity::class.java)
            intent.putExtra(RouteActivity.EXTRA_COMPANY, position + 1)
            startActivity(intent)
        }
    }
}
